enhancements/node-tuning/mixed-cpu-node-plugin.md

```diff
@@ -0,0 +1,353 @@
+
+There is a plan for a feature called [shared-partition](https://github.com/openshift/enhancements/pull/1421) 
+that implements additional pool.
+After the shared-partition lands, the pool layout should look roughly like:
+1. `reservedSystemCpus`
+2. `guaranteedCpus` <- new pool
+3. `sharedCpus` = all CPUs - `reservedSystemCpus` - `guaranteedCpus`
+
+And mixed-cpu-node-plugin will allocate the shared cpus from the `sharedCpus` pool and not
+from the reservedSystemCpus anymore.
+
+The node-plugin populates a special device named `openshift.io/shared-cpus` to provide way for pods to request
+for this special type of CPUs.
+There's no meaning to the value/number of `openshift.io/shared-cpus` devices that the pod requests.
+For example:
+```yaml
+requests:
+   openshift.io/shared-cpus: 1
+```
+Or
+```yaml
+requests:
+   openshift.io/shared-cpus: 2
+```
+Are the same in that context, the container has access to the same number of shared-cpus in both cases.
+In other words, the device request only uses as a hint to tell the plugin that shared-cpus required for the container and
+does not indicate any actual value.
+
+The reason for specifying a device, and not, for example, an annotation,
+is because when application pod requests for a device, the scheduler makes the pod pending till device up and running.
+This gives the node plugin room for setup and not being depended on pod admission order.
+So only when the node plugin finishes the registration process with NRI and device plugin, the scheduler
+admits the application pod.
+
+Once a container under the pod requests and gets access to shared cpus, new environment variable named
+`OPENSHIFT_SHARED_CPUS=<CPU-IDs>` will be present under the container's environment variables.
+
+The mixed-cpu-node-plugin itself is essentially a pod that runs as a DaemonSet on the worker nodes.
+cluster-node-tuning-operator (NTO) will be modified to support the deployment of the DaemonSet and additional
+resources (ServiceAccount, RBAC resources, SecurityContextConstraint, etc.)
+NTO will also be responsible to watch, monitor and report the node-plugin's DaemonSet status.
+Additionally, since the plugin is part of OCP infrastructure pods, NTO annotates the pods with:
+```yaml
+target.workload.openshift.io/management: {
+  "effect":"PreferredDuringScheduling"
+}
+```
+See https://github.com/openshift/enhancements/blob/master/enhancements/workload-partitioning/management-workload-partitioning.md for details.
+
+Since both the components are related,
+mixed-cpu-node-plugin [source code](https://github.com/openshift-kni/mixed-cpu-node-plugin) would be vendor under NTO.
+By having different main.go + container entrypoint, NTO image can be also used as the
+mixed-cpu-node-plugin image. 
+This way saves the overhead of having a separate container image with all the maintenance cost that comes with it.
+
+In order to activate NRI in container-runtime, NTO creates MachineConfig to enable NRI in 
```

Comment:
ideally the proposal would mention the mc that'll be used. as a note: I think we should take care to have it only consist of 
```
[crio.nri]
enable_nri = true
```
as we imagine a future where more extensions to openshift use NRI, we'll want to make sure they don't set on each others toes, so this should be the only config field we change for it (i.e: don't change the location of the listen socket or plugin dir)

Reply:
Ack i'll add it. 
BTW this is exactly what I did: https://github.com/Tal-or/cluster-node-tuning-operator/blob/mixed_cpu_plugin_deployment_support/assets/performanceprofile/configs/99-nri.conf